File: src/utils/project.js
const fs = require("fs-extra");
const path = require("path");
const os = require("os");
const chalk = require("chalk");

class ProjectManager {
  constructor() {
    this.configFile = "writers.config.json";
    this.baseDir = process.env.WRITERS_PROJECT_DIR || process.cwd();
    this.projectStructure = {
      chapters: "chapters",
      scenes: "scenes",
      characters: "characters",
      notes: "notes",
      shortstories: "shortstories",
      stories: "stories",
      drafts: "drafts",
      "submission-ready": "submission-ready",
      exercises: "exercises",
      prompts: "prompts",
      exports: "exports",
    };
  }

  /**
   * Resolve a path relative to the current project base directory
   */
  resolvePath(...segments) {
    return path.resolve(this.baseDir, ...segments);
  }

  /**
   * Set the project base directory at runtime (e.g., from GUI)
   */
  setBaseDir(dir) {
    if (dir && typeof dir === "string") {
      this.baseDir = dir;
    }
  }

  /**
   * Check if current directory is a writers project
   */
  isWritersProject() {
    return fs.existsSync(this.resolvePath(this.configFile));
  }

  /**
   * Get project configuration
   */
  async getConfig() {
    if (!this.isWritersProject()) {
      throw new Error(
        'Not a writers project. Run "writers init" to initialize.',
      );
    }

    try {
      const config = await fs.readJson(this.resolvePath(this.configFile));
      return config;
    } catch (error) {
      throw new Error("Failed to read project configuration.");
    }
  }

  /**
   * Update project configuration
   */
  async updateConfig(updates) {
    const config = await this.getConfig();
    const newConfig = { ...config, ...updates };
    await fs.writeJson(this.resolvePath(this.configFile), newConfig, {
      spaces: 2,
    });
    return newConfig;
  }

  /**
   * Create project structure
   */
  async createProjectStructure() {
    for (const dir of Object.values(this.projectStructure)) {
      await fs.ensureDir(this.resolvePath(dir));
    }
  }

  /**
   * Initialize a new project
   */
  async initProject(options = {}) {
    if (this.isWritersProject()) {
      throw new Error("Already a writers project.");
    }

    const config = {
      name: options.name || path.basename(process.cwd()),
      author: options.author || "Unknown Author",
      created: new Date().toISOString(),
      version: "1.0.0",
      type: options.type || "novel",
      wordGoal: options.wordGoal || 50000,
      settings: {
        autoSave: true,
        backups: true,
        defaultEditor: options.editor || "nano",
      },
      structure:
        options.type === "blog"
          ? {
              posts: "posts",
              notes: "notes",
              exports: "exports",
              assets: "assets/images",
              templates: "templates",
            }
          : this.projectStructure,
    };

    // If blog project, override internal projectStructure so later calls use it
    if (options.type === "blog") {
      this.projectStructure = {
        posts: "posts",
        notes: "notes",
        exports: "exports",
        assets: "assets/images",
        templates: "templates",
      };
    }

    await fs.writeJson(this.resolvePath(this.configFile), config, {
      spaces: 2,
    });
    await this.createProjectStructure();

    // Create initial README
    const readme = this.generateReadme(config);
    await fs.writeFile(this.resolvePath("README.md"), readme);

    return config;
  }

  /**
   * Generate project README
   */
  generateReadme(config) {
    if ((config.type || "").toLowerCase() === "blog") {
      return `# ${config.name}

**Author:** ${config.author}
**Created:** ${new Date(config.created).toLocaleDateString()}
**Type:** Blog
**Word Goal:** ${config.wordGoal.toLocaleString()} words

Welcome to your blog project. Use the GUI or CLI to create and manage posts.

## Project Structure

- \`posts/\` - Your blog posts (Markdown). Each file represents a post.
- \`notes/\` - Ideas, outlines, research, and drafts not yet promoted to posts.
- \`assets/images/\` - Images and media you embed in posts.
- \`templates/\` - (Optional) Content or metadata templates.
- \`exports/\` - Generated/exported versions (HTML, PDF, etc.).

## Getting Started

1. Create a new post: \`writers new post "My First Post"\` or click "New Post" in the GUI.
2. Draft and edit in the GUI editor.
3. Export when ready.

Happy blogging!
`;
    }

    return `# ${config.name}

**Author:** ${config.author}
**Created:** ${new Date(config.created).toLocaleDateString()}
**Word Goal:** ${config.wordGoal.toLocaleString()} words

## Project Structure

- \`chapters/\` - Main story chapters
- \`scenes/\` - Individual scenes and drafts
- \`characters/\` - Character profiles and development
- \`notes/\` - Research, plot notes, and ideas
- \`shortstories/\` - Complete short stories
- \`exports/\` - Exported versions of your work

## Quick Start

- **Create a new chapter:** \`writers new chapter "Chapter Title"\`
- **Create a short story:** \`writers new shortstory "Story Title"\`
- **Start writing:** \`writers write chapter1\`
- **View statistics:** \`writers stats\`
- **Export your work:** \`writers export html\`

## Writing Progress

Use \`writers stats\` to track your progress toward your ${config.wordGoal.toLocaleString()} word goal.

---

*This project was created with Writers CLI - A tool for novelists and short story writers*
`;
  }

  /**
   * Get all files of a specific type
   */
  async getFiles(type) {
    const dir = this.projectStructure[type] || type;
    if (!(await fs.pathExists(this.resolvePath(dir)))) {
      return [];
    }

    const files = await fs.readdir(this.resolvePath(dir));
    const fileStats = await Promise.all(
      files.map(async (file) => {
        const filePath = this.resolvePath(dir, file);
        try {
          const stats = await fs.stat(filePath);
          return {
            name: path.basename(file, path.extname(file)),
            path: filePath,
            fullName: file,
            isDirectory: stats.isDirectory(),
            size: stats.size,
            modified: stats.mtime,
            created: stats.birthtime,
          };
        } catch (error) {
          console.error(`Error getting stats for ${filePath}:`, error);
          return null;
        }
      }),
    );

    // Filter out any null entries and directories, and sort by name
    return fileStats
      .filter((file) => file && !file.isDirectory)
      .sort((a, b) => a.name.localeCompare(b.name));
  }

  /**
   * Create a new file with template
   */
  async createFile(type, name, template = "") {
    const dir = this.projectStructure[type] || type;

    await fs.ensureDir(this.resolvePath(dir));
    const fileName = `${this.sanitizeFileName(name)}.md`;
    const filePath = this.resolvePath(dir, fileName);

    if (await fs.pathExists(filePath)) {
      throw new Error(`${type} "${name}" already exists.`);
    }

    const content = this.generateTemplate(type, name, template);
    await fs.writeFile(filePath, content);

    return {
      name,
      path: filePath,
      type,
    };
  }

  /**
   * Generate content template based on type
   */
  generateTemplate(type, name, customTemplate = "") {
    if (customTemplate) {
      return customTemplate;
    }

    const timestamp = new Date().toISOString().split("T")[0];

    switch (type) {
      case "chapters":
        return `# ${name}

*Created: ${timestamp}*

---

## Summary
Brief summary of what happens in this chapter.

## Notes
- Plot points
- Character development
- Important details

---

## Content

Write your chapter content here...

`;

      case "scenes":
        return `# ${name}

*Created: ${timestamp}*

**Setting:** Describe the location and time
**Characters:** List main characters in this scene
**Purpose:** What this scene accomplishes

---

Write your scene here...

`;

      case "characters":
        return `# ${name}

*Created: ${timestamp}*

## Basic Information
- **Full Name:**
- **Age:**
- **Occupation:**
- **Location:**

## Physical Description
- **Appearance:**
- **Distinguishing Features:**

## Personality
- **Traits:**
- **Motivations:**
- **Fears:**
- **Flaws:**

## Background
- **History:**
- **Family:**
- **Important Events:**

## Role in Story
- **Purpose:**
- **Character Arc:**
- **Relationships:**

## Notes
Additional notes and development ideas...

`;

      case "shortstories":
      case "stories":
        return `# ${name}

*Created: ${timestamp}*

---

## Story Information
- **Genre:**
- **Target Length:** (words)
- **Theme:**
- **Setting:**
- **Status:** Planning

## Summary
Brief one-sentence summary of the story.

## Characters
- **Protagonist:**
- **Supporting Characters:**

## Plot Structure
- **Opening:** How the story begins
- **Conflict:** Central problem or tension
- **Climax:** Peak moment
- **Resolution:** How it ends

## Notes
Story ideas, themes, and development notes...

---

## Story Content

Write your short story here...

`;

      case "drafts":
        return `# ${name} - Draft

*Created: ${timestamp}*

---

## Draft Information
- **Version:** 1.0
- **Status:** Drafting
- **Target Length:**
- **Genre:**

## Draft Notes
[What are you focusing on in this draft?]

## Story Content

Write your draft here...

`;

      case "exercises":
        return `# Writing Exercise: ${name}

*Created: ${timestamp}*

---

## Exercise Information
- **Skill Focus:** [What are you practicing?]
- **Time Limit:** [How long to spend?]
- **Success Criteria:** [How will you know you succeeded?]

## Exercise Content

[Begin your writing exercise here...]

## Reflection
[What did you learn from this exercise?]

`;

      case "prompts":
        return `# Prompt Response: ${name}

*Created: ${timestamp}*

---

## Prompt Information
- **Original Prompt:** [Write the prompt here]
- **Time Limit:**
- **Target Length:**

## Response

[Write your response here...]

`;

      case "submission-ready":
        return `# ${name}

*Created: ${timestamp}*
*Status: Submission Ready*

---

Write your polished story here...

`;

      default:
        return `# ${name}

*Created: ${timestamp}*

---

`;
    }
  }

  /**
   * Sanitize filename
   */
  sanitizeFileName(name) {
    return name
      .toLowerCase()
      .replace(/[^a-z0-9\s-]/g, "")
      .replace(/\s+/g, "-")
      .replace(/-+/g, "-")
      .trim("-");
  }

  /**
   * Get project statistics
   */
  async getProjectStats() {
    const config = await this.getConfig();
    const isBlog = (config.type || "").toLowerCase() === "blog";
    const stats = {
      project: config.name,
      author: config.author,
      created: config.created,
      wordGoal: config.wordGoal,
      chapters: [],
      totalWords: 0,
      totalCharacters: 0,
      files: isBlog
        ? {
            posts: 0,
            notes: 0,
          }
        : {
            chapters: 0,
            scenes: 0,
            characters: 0,
            notes: 0,
            shortstories: 0,
          },
    };

    // Count files and words for each type
    for (const [type, dir] of Object.entries(this.projectStructure)) {
      if (type === "exports") continue;

      try {
        const files = await this.getFiles(type);
        stats.files[type] = files.length;

        for (const file of files) {
          try {
            const content = await fs.readFile(file.path, "utf8");
            const words = this.countWords(content);
            const characters = content.length;

            stats.totalWords += words;
            stats.totalCharacters += characters;

            if (type === "chapters") {
              stats.chapters.push({
                name: file.name,
                words,
                characters,
                path: file.path,
              });
            }
          } catch (error) {
            console.warn(chalk.yellow(`Warning: Could not read ${file.path}`));
          }
        }
      } catch (error) {
        // Directory might not exist, initialize with 0
        stats.files[type] = 0;
      }
    }

    stats.progress = Math.min(100, (stats.totalWords / stats.wordGoal) * 100);

    return stats;
  }

  /**
   * Count words in text
   */
  countWords(text) {
    // Remove markdown formatting and count words
    const cleanText = text
      .replace(/^#+ .*/gm, "") // Remove headers
      .replace(/\*\*(.*?)\*\*/g, "$1") // Remove bold
      .replace(/\*(.*?)\*/g, "$1") // Remove italic
      .replace(/`(.*?)`/g, "$1") // Remove code
      .replace(/\[(.*?)\]\(.*?\)/g, "$1") // Remove links
      .replace(/^\s*[-*+]\s+/gm, "") // Remove list markers
      .replace(/^\s*\d+\.\s+/gm, "") // Remove numbered lists
      .replace(/^\s*>/gm, "") // Remove blockquotes
      .replace(/---+/g, "") // Remove horizontal rules
      .trim();

    if (!cleanText) return 0;

    return cleanText.split(/\s+/).filter((word) => word.length > 0).length;
  }

  /**
   * Backup project
   */
  async createBackup() {
    const timestamp = new Date().toISOString().replace(/[:.]/g, "-");
    const backupDir = `backups/backup-${timestamp}`;

    await fs.ensureDir("backups");
    await fs.copy(".", backupDir, {
      filter: (src) => {
        const relativePath = path.relative(".", src);
        return (
          !relativePath.startsWith("backups") &&
          !relativePath.startsWith("node_modules") &&
          !relativePath.startsWith(".git")
        );
      },
    });

    return backupDir;
  }

  /**
   * Export project structure as JSON
   */
  async exportStructure() {
    const stats = await this.getProjectStats();
    const structure = {};

    // Copy project structure as JSON
    for (const [type, dir] of Object.entries(this.projectStructure)) {
      if (type === "exports") continue;

      try {
        const files = await this.getFiles(type);
        structure[type] = [];

        for (const file of files) {
          try {
            const content = await fs.readFile(file.path, "utf8");
            structure[type].push({
              name: file.name,
              path: file.path,
              words: this.countWords(content),
              content: content,
            });
          } catch (error) {
            console.warn(chalk.yellow(`Warning: Could not read ${file.path}`));
          }
        }
      } catch (error) {
        // Directory might not exist, initialize empty array
        structure[type] = [];
      }
    }

    return {
      metadata: {
        project: stats.project,
        author: stats.author,
        created: stats.created,
        exported: new Date().toISOString(),
        totalWords: stats.totalWords,
        wordGoal: stats.wordGoal,
        progress: stats.progress,
      },
      structure,
    };
  }
}

module.exports = new ProjectManager();

File: go-port/internal/project/project.go
package project

import (
	"fmt"
	"os"
	"path/filepath"
	"time"
)

// Config holds project configuration
type Config struct {
	Name        string
	Type        string
	Template    string
	Description string
	GitInit     bool
}

// Project represents a writing project
type Project struct {
	Name        string    `yaml:"name"`
	Type        string    `yaml:"type"`
	Description string    `yaml:"description"`
	Author      string    `yaml:"author"`
	Created     time.Time `yaml:"created"`
	Updated     time.Time `yaml:"updated"`
	Version     string    `yaml:"version"`
	Goals       Goals     `yaml:"goals"`
	Settings    Settings  `yaml:"settings"`
}

// Goals holds project goals and targets
type Goals struct {
	DailyWords  int `yaml:"daily_words"`
	TotalWords  int `yaml:"total_words"`
	Deadline    string `yaml:"deadline"`
	Description string `yaml:"description"`
}

// Settings holds project-specific settings
type Settings struct {
	Theme         string `yaml:"theme"`
	AutoSave      bool   `yaml:"auto_save"`
	WordWrap      bool   `yaml:"word_wrap"`
	ShowWordCount bool   `yaml:"show_word_count"`
	BackupEnabled bool   `yaml:"backup_enabled"`
}

// Initialize creates a new writing project
func Initialize(projectPath string, config *Config) error {
	// Create project directory if it doesn't exist
	if err := os.MkdirAll(projectPath, 0755); err != nil {
		return fmt.Errorf("failed to create project directory: %w", err)
	}

	// Create project structure based on type
	if err := createProjectStructure(projectPath, config.Type); err != nil {
		return fmt.Errorf("failed to create project structure: %w", err)
	}

	// Create project configuration file
	project := &Project{
		Name:        config.Name,
		Type:        config.Type,
		Description: config.Description,
		Author:      getAuthorName(),
		Created:     time.Now(),
		Updated:     time.Now(),
		Version:     "1.0.0",
		Goals: Goals{
			DailyWords:  500,
			TotalWords:  50000,
			Description: "Complete first draft",
		},
		Settings: Settings{
			Theme:         "dark",
			AutoSave:      true,
			WordWrap:      false,
			ShowWordCount: true,
			BackupEnabled: true,
		},
	}

	if err := saveProjectConfig(projectPath, project); err != nil {
		return fmt.Errorf("failed to save project config: %w", err)
	}

	// Initialize git repository if requested
	if config.GitInit {
		if err := initializeGit(projectPath); err != nil {
			return fmt.Errorf("failed to initialize git: %w", err)
		}
	}

	return nil
}

// createProjectStructure creates the directory structure for different project types
func createProjectStructure(projectPath, projectType string) error {
	structures := map[string][]string{
		"novel": {
			"chapters",
			"characters",
			"notes",
			"research",
			"drafts",
			"outline",
		},
		"shortstories": {
			"stories",
			"drafts",
			"published",
			"submissions",
			"notes",
		},
		"article": {
			"drafts",
			"research",
			"images",
			"notes",
		},
		"screenplay": {
			"scenes",
			"characters",
			"treatments",
			"notes",
			"drafts",
		},
		"poetry": {
			"poems",
			"collections",
			"drafts",
			"notes",
		},
		"journal": {
			"entries",
			"templates",
			"notes",
		},
		"academic": {
			"chapters",
			"references",
			"research",
			"notes",
			"drafts",
			"figures",
		},
	}

	dirs, exists := structures[projectType]
	if !exists {
		dirs = structures["novel"] // default structure
	}

	// Create directories
	for _, dir := range dirs {
		dirPath := filepath.Join(projectPath, dir)
		if err := os.MkdirAll(dirPath, 0755); err != nil {
			return err
		}

		// Create a README in each directory
		readmePath := filepath.Join(dirPath, "README.md")
		readmeContent := fmt.Sprintf("# %s\n\nThis directory contains %s for your %s project.\n",
			dir, dir, projectType)

		if err := os.WriteFile(readmePath, []byte(readmeContent), 0644); err != nil {
			return err
		}
	}

	// Create main project files
	if err := createProjectFiles(projectPath, projectType); err != nil {
		return err
	}

	return nil
}

// createProjectFiles creates initial files for the project
func createProjectFiles(projectPath, projectType string) error {
	files := map[string]string{
		"README.md": getReadmeTemplate(projectType),
		".gitignore": getGitignoreTemplate(),
	}

	// Add project-specific files
	switch projectType {
	case "novel":
		files["outline/main-outline.md"] = getNovelOutlineTemplate()
		files["characters/character-template.md"] = getCharacterTemplate()
		files["chapters/chapter-01.md"] = getChapterTemplate()
	case "shortstories":
		files["stories/story-template.md"] = getStoryTemplate()
		files["submissions/submission-tracker.md"] = getSubmissionTrackerTemplate()
	case "article":
		files["article.md"] = getArticleTemplate()
	case "screenplay":
		files["screenplay.fountain"] = getScreenplayTemplate()
		files["characters/character-list.md"] = getCharacterTemplate()
	case "poetry":
		files["poems/first-poem.md"] = getPoemTemplate()
		files["collections/collection-notes.md"] = getCollectionTemplate()
	case "journal":
		files["entries/template.md"] = getJournalTemplate()
	case "academic":
		files["thesis.md"] = getThesisTemplate()
		files["references/bibliography.md"] = getBibliographyTemplate()
	}

	// Create files
	for filePath, content := range files {
		fullPath := filepath.Join(projectPath, filePath)

		// Create directory if it doesn't exist
		if err := os.MkdirAll(filepath.Dir(fullPath), 0755); err != nil {
			return err
		}

		if err := os.WriteFile(fullPath, []byte(content), 0644); err != nil {
			return err
		}
	}

	return nil
}

// saveProjectConfig saves the project configuration
func saveProjectConfig(projectPath string, project *Project) error {
	// For now, create a simple YAML-like file
	// In a full implementation, you'd use gopkg.in/yaml.v3
	configPath := filepath.Join(projectPath, ".writers-project.yml")

	content := fmt.Sprintf(`name: %s
type: %s
description: %s
author: %s
created: %s
updated: %s
version: %s

goals:
  daily_words: %d
  total_words: %d
  deadline: "%s"
  description: "%s"

settings:
  theme: %s
  auto_save: %t
  word_wrap: %t
  show_word_count: %t
  backup_enabled: %t
`,
		project.Name,
		project.Type,
		project.Description,
		project.Author,
		project.Created.Format(time.RFC3339),
		project.Updated.Format(time.RFC3339),
		project.Version,
		project.Goals.DailyWords,
		project.Goals.TotalWords,
		project.Goals.Deadline,
		project.Goals.Description,
		project.Settings.Theme,
		project.Settings.AutoSave,
		project.Settings.WordWrap,
		project.Settings.ShowWordCount,
		project.Settings.BackupEnabled,
	)

	return os.WriteFile(configPath, []byte(content), 0644)
}

// initializeGit initializes a git repository
func initializeGit(projectPath string) error {
	// This is a simplified implementation
	// In a real implementation, you'd use go-git or exec git commands
	gitPath := filepath.Join(projectPath, ".git")
	return os.MkdirAll(gitPath, 0755)
}

// getAuthorName gets the author name from environment or defaults
func getAuthorName() string {
	if name := os.Getenv("GIT_AUTHOR_NAME"); name != "" {
		return name
	}
	if name := os.Getenv("USER"); name != "" {
		return name
	}
	return "Author"
}

// Template functions
func getReadmeTemplate(projectType string) string {
	return fmt.Sprintf(`# %s Project

This is a %s project created with Writers CLI.

## Getting Started

Use the Writers CLI editor to begin writing:

` + "```" + `bash
writers edit
` + "```" + `

## Project Structure

This project follows the standard %s structure with organized directories for different aspects of your work.

## Goals

- Set your daily writing goals
- Track your progress
- Meet your deadlines

## Tips

- Use F2 to switch between dark and light themes
- Press F9 for typewriter mode
- Use Ctrl+W to check word count

Happy writing! 📝
`, projectType, projectType, projectType)
}

func getGitignoreTemplate() string {
	return `# Writers CLI
.writers-backup/
*.tmp
*.bak

# OS
.DS_Store
Thumbs.db

# Editor
*.swp
*.swo
*~

# Compiled
*.pdf
*.docx
*.epub
*.mobi
`
}

func getNovelOutlineTemplate() string {
	return `# Novel Outline

## Story Overview

**Title:** Your Novel Title
**Genre:**
**Target Length:** 80,000 words
**Logline:** One sentence summary of your story

## Three-Act Structure

### Act I - Setup (25%)
- **Hook:**
- **Inciting Incident:**
- **Plot Point 1:**

### Act II - Confrontation (50%)
- **Rising Action:**
- **Midpoint:**
- **Plot Point 2:**

### Act III - Resolution (25%)
- **Climax:**
- **Falling Action:**
- **Resolution:**

## Characters

- **Protagonist:**
- **Antagonist:**
- **Supporting Characters:**

## Themes

- Primary theme:
- Secondary themes:

## Chapter Breakdown

1. Chapter 1 -
2. Chapter 2 -
3. Chapter 3 -
...
`
}

func getCharacterTemplate() string {
	return `# Character Profile

## Basic Information

**Name:**
**Age:**
**Occupation:**
**Location:**

## Physical Description

**Appearance:**
**Distinguishing Features:**

## Personality

**Traits:**
**Strengths:**
**Weaknesses:**
**Fears:**
**Goals:**

## Background

**Family:**
**Education:**
**Important Events:**

## Role in Story

**Function:**
**Character Arc:**
**Relationships:**

## Voice and Dialogue

**Speaking Style:**
**Catchphrases:**
**Internal Voice:**

## Notes

Additional character notes and development ideas.
`
}

func getChapterTemplate() string {
	return `# Chapter 1

## Chapter Summary

Brief summary of what happens in this chapter.

## Goals

- What needs to be accomplished
- Character development
- Plot advancement

---

Start writing your chapter here...

## Notes

- Revision notes
- Ideas for improvement
- Continuity reminders
`
}

func getStoryTemplate() string {
	return `# Story Title

**Author:** Your Name
**Word Count:** 0 / Target: 5,000
**Status:** Draft
**Genre:**
**Theme:**

## Story Summary

Brief summary of your story.

## Notes

- Character notes
- Plot ideas
- Setting details

---

## Story

Start your story here...
`
}

func getSubmissionTrackerTemplate() string {
	return `# Submission Tracker

| Story | Market | Submitted | Response | Status | Notes |
|-------|--------|-----------|----------|---------|-------|
| Story 1 | Magazine A | 2024-01-01 | - | Pending | - |
| Story 2 | Magazine B | 2024-01-15 | Rejection | Closed | Good feedback |

## Markets to Consider

- **Magazine A** - Sci-fi, pays $0.10/word, 5,000 word limit
- **Magazine B** - Literary, pays $50, 3,000 word limit
- **Magazine C** - Horror, pays $25, 2,500 word limit

## Submission Guidelines

- Always read guidelines carefully
- Track response times
- Keep records of feedback
- Multiple submissions only if allowed
`
}

func getArticleTemplate() string {
	return `# Article Title

**Author:** Your Name
**Publication:** Target Publication
**Word Count:** 0 / Target: 1,500
**Status:** Draft
**Deadline:**

## Article Outline

### Introduction
- Hook
- Background
- Thesis

### Main Points
1. First point
2. Second point
3. Third point

### Conclusion
- Summary
- Call to action

---

## Article

Start writing your article here...

## Research Notes

- Sources
- Statistics
- Quotes
- References
`
}

func getScreenplayTemplate() string {
	return `Title: YOUR SCREENPLAY TITLE
Author: Your Name
Contact: your.email@example.com

FADE IN:

EXT. LOCATION - DAY

Write your screenplay here using standard formatting.

Character dialogue should be centered and formatted properly.

FADE OUT.
`
}

func getPoemTemplate() string {
	return `# First Poem

**Title:** Untitled
**Form:** Free verse
**Theme:**
**Date:** Today

---

Write your poem here.
Each line can be its own line,
or you can group them into stanzas.

Let your creativity flow.

---

## Notes

- Inspiration
- Revision ideas
- Form experiments
`
}

func getCollectionTemplate() string {
	return `# Collection Notes

**Collection Title:**
**Theme:**
**Target Length:** 20-30 poems
**Status:** Planning

## Poems to Include

1. Poem 1 - Complete
2. Poem 2 - Draft
3. Poem 3 - Idea

## Themes and Motifs

- Primary theme:
- Secondary themes:
- Recurring images:

## Structure and Flow

- Opening poem:
- Sections:
- Closing poem:

## Submission Plans

- Journals to consider:
- Contest deadlines:
- Reading opportunities:
`
}

func getJournalTemplate() string {
	return `# Journal Entry - {{DATE}}

## Mood

**Overall feeling:**
**Energy level:**

## Today's Events

What happened today that was significant?

## Thoughts and Reflections

What's on your mind?

## Gratitude

Three things you're grateful for:
1.
2.
3.

## Tomorrow's Goals

What do you want to accomplish tomorrow?

---

## Free Writing

Just write whatever comes to mind...
`
}

func getThesisTemplate() string {
	return `# Thesis Title

**Author:** Your Name
**Department:**
**Advisor:**
**Target Length:** 100-200 pages
**Defense Date:**

## Abstract

Brief summary of your research and findings.

## Table of Contents

1. Introduction
2. Literature Review
3. Methodology
4. Results
5. Discussion
6. Conclusion
7. References
8. Appendices

---

## Chapter 1: Introduction

Start your thesis here...

### Research Question

What question are you trying to answer?

### Hypothesis

What do you expect to find?

### Significance

Why is this research important?
`
}

func getBibliographyTemplate() string {
	return `# Bibliography

## Primary Sources

- Author, A. (Year). *Title of work*. Publisher.
- Author, B. (Year). Title of article. *Journal Name*, Volume(Issue), pages.

## Secondary Sources

- Author, C. (Year). *Title of book*. Publisher.
- Author, D. (Year). Title of article. *Journal Name*, Volume(Issue), pages.

## Online Sources

- Author, E. (Year). Title of webpage. *Website Name*. URL

## Guidelines

- Follow your institution's citation style (APA, MLA, Chicago, etc.)
- Keep track of all sources as you research
- Include page numbers for direct quotes
- Verify all URLs and access dates
`
}
